File: csharp/Monopoly/Monopoly/RLClasses/EligibilityTrace.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Monopoly.RLClasses
{
    [Serializable]
    public class EligibilityTrace
    {
        public Observation observation { get; set; }
        public Action action { get; set; }
        public double value { get; set; }

        public EligibilityTrace() { }

        public EligibilityTrace(Observation o, Action a, double v)
        {
            this.observation = o;
            this.action = a;
            this.value = v;
        }
    }
}

File: java/src/main/java/org/dinbur/ml/rl/monopoly/state/EligibilityTrace.java
package org.dinbur.ml.rl.monopoly.state;

public class EligibilityTrace {

    private final Observation observation;
    private final Action action;
    private final double reward;

    public EligibilityTrace(final Observation observation, final Action action, final double reward) {
        this.observation = observation;
        this.action = action;
        this.reward = reward;
    }

    public Observation getObservation() {
        return observation;
    }

    public Action getAction() {
        return action;
    }

    public double getReward() {
        return reward;
    }
}
